components/server/src/prometheus-metrics.ts

```diff
@@ -65,3 +65,26 @@ const apiCallUserCounter = new prometheusClient.Counter({
 export function increaseApiCallUserCounter(method: string, user: string) {
     apiCallUserCounter.inc({ method, user });
 }
+
+const httpRequestTotal = new prometheusClient.Counter({
+    name: 'gitpod_server_http_requests_total',
+    help: 'Total amount of HTTP requests per express route',
+    labelNames: ['method', 'route', 'statusCode'],
+    registers: [prometheusClient.register],
+});
+
+export function increaseHttpRequestCounter(method: string, route: string, statusCode: number) {
+    httpRequestTotal.inc({ method, route, statusCode });
+}
+
+const httpRequestDuration = new prometheusClient.Histogram({
+    name: 'gitpod_server_http_request_duration_seconds',
+    help: 'Duration of HTTP requests in seconds',
+    labelNames: ['method', 'route', 'statusCode'],
```

Comment:
what is the difference between `method` and `route`? 🤔

Reply:
method is an http method, e.g. `GET`
route is an express route, e.g. `/code-sync/resource/:kind/:rev`